### PR TITLE
Update WebFetch, and its users, to use the new service hostnames and endpoints, as well as HTTPS.

### PR DESCRIPTION
**Pull request by mkrautz**

### CODE REVIEW

**Comment by mkrautz** [comment]
WIP

**Comment by mkrautz** [comment]
PTAL

File: src/mumble/Plugins.cpp [hunk 1]
```diff
@@ -540,7 +540,7 @@ void Plugins::fetched(QByteArray data, QUrl url) {
 
 	bool rescan = false;
 	const QString &urlPath = url.path();
-	if (urlPath == QLatin1String("/plugins.php")) {
+	if (urlPath == QLatin1String("/v1/pa-plugins")) {
```
Comment (hacst) [inline src/mumble/Plugins.cpp]: We should take this chance to split up the fetched slot function used for the list being returned and a download being finished. E.g. ```fetched_update_pa_plugins``` and ```fetched_pa_plugin_dl``` or something.

File: src/mumble/WebFetch.cpp [hunk 2]
```diff
@@ -35,19 +35,36 @@
...
 	qnr = Network::get(url);
 	connect(qnr, SIGNAL(finished()), this, SLOT(finished()));
 	connect(this, SIGNAL(fetched(QByteArray,QUrl,QMap<QString,QString>)), obj, slot);
 }
 
+QString WebFetch::prefixedServiceHost() const {
+	if (g.s.qsServicePrefix.isEmpty()) {
+		return serviceHost();
+	}
+	return QString::fromLatin1("%1-%2.mumble.info").arg(g.s.qsServicePrefix, m_service);
```
Comment (hacst) [inline src/mumble/WebFetch.cpp]: Maybe we should do "%2%1" instead. That way we can decide on the service side whether we want ```publist.foo.mumble.info``` or ```publist-foo.mumble.info```. WDYT?

File: src/mumble/WebFetch.cpp [hunk 3]
```diff
@@ -76,20 +93,15 @@ void WebFetch::finished() {
 			QString value = fromUtf8(qnr->rawHeader(headerName));
 			if (! name.isEmpty() && ! value.isEmpty()) {
 				headers.insert(name, value);
-				if (name == QLatin1String("Geo-Country-Code"))
-					g.s.qsRegionalHost = value.toLower() + QLatin1String(".mumble.info");
+				if (name == QLatin1String("Use-Service-Prefix"))
```
Comment (hacst) [inline src/mumble/WebFetch.cpp]: Depending on how we decide it's not exactly a prefix anymore ;)

File: src/mumble/WebFetch.h [hunk 4]
```diff
@@ -52,14 +52,18 @@ class WebFetch : public QObject {
...
+		QString serviceHost() const;
+
+		WebFetch(QString service, QUrl url, QObject *obj, const char *slot);
 	signals:
 		void fetched(QByteArray data, QUrl url, QMap<QString, QString> headers);
 	protected slots:
 		void finished();
 	public:
-		static void fetch(const QUrl &url, QObject *obj, const char *slot);
+		static void fetch(const QString &service, const QUrl &url, QObject *obj, const char *slot);
```
Comment (hacst) [inline src/mumble/WebFetch.h]: Should be documented

File: src/mumble/Settings.h [hunk 5]
```diff
@@ -303,7 +303,7 @@ struct Settings {
 	ProxyType ptProxyType;
 	QString qsProxyHost, qsProxyUsername, qsProxyPassword;
 	unsigned short usProxyPort;
-	QString qsRegionalHost;
+	QString qsServicePrefix;
```
Comment (hacst) [inline src/mumble/Settings.h]: Needs documentation.

**Comment by hacst** [comment]
See comments. Apart from that LGTM.

**Comment by mkrautz** [comment]
@hacst PTAL at v2: https://github.com/mkrautz/mumble/commit/3659c14f73fdaa251ea30a169ee181c8cc7f75ef

**Comment by hacst** [comment]
V2 LGTM (see comments).

**Comment by mkrautz** [comment]
Landed v2 as acb69be2425d59e041975f0a04c63fcfbdc73b0d.

**Pull request closed without merging**